components/ca-list.js

```diff
@@ -0,0 +1,722 @@
+define([
+    './helpers/component-support.js',
+    './helpers/get-sorted-schema-keys.js',
+    './helpers/create-element-legacy.js',
+    './helpers/trace.js',
+    './helpers/clear-element.js',
+    './helpers/parent-by-attribute.js',
+    './helpers/parse-iso-local-date.js',
+    './helpers/secureajax.js',
+    'document-register-element'
+], (componentSupport, getSortedSchemaKeys, createElement, trace, clearElement, getParentByAttribute, parseIsoLocalDate, ajax) => {
+    /**
+     * @exports ca-list
+     * A custom HTML element (Web Component) that can be created using document.createElement('ca-list') or included in a HTML page as an element.
+     * Setting the .src attribute forces the component to load the external (JSON v3 Schema/JSON data object) url,
+     * follow the HATEOAS links to locate either the schema or data & render a HTML table matching the schema/data.
+     */
+    class List extends HTMLElement {
+        /**
+         * Executes when the element is first created
+         * @returns {undefined} nothing
+         */
+        createdCallback() {
+
+            // always render the search filter, but hide it via css unless data-searchable=true
+            this.filterEl = createElement(this, 'input', {
+                class: 'ca-list-filter',
+                type: 'text',
+                placeholder: 'Enter search terms to filter the list below'
+            });
+            this.filterEl.onkeyup = this.eventDelegate.bind(this);
+
+            // build the HTML table and keep a reference to the elements we need to update later
+            this.table = createElement(this, 'table', {});
+            this.colgroup = createElement(this.table, 'colgroup', {});
+            this.thead = createElement(this.table, 'thead', {});
+            this.tbody = createElement(this.table, 'tbody', {});
+
+            // capture click events and map them to internal UI event handler (sits at parent level and catches child clicks)
+            this.table.onclick = this.eventDelegate.bind(this);
+
+            // react to literal attributes
+            if (this.src !== '') this.loadSrc();
+            if (this.schema) this.createTable();
+            if (this.sortIndex > -1) this.sortTable();
+        }
+
+        // noinspection JSUnusedGlobalSymbols
+        /**
+         * Executes when the element is attached to the DOM
+         * @returns {undefined} nothing
+         */
+        attachedCallback() {
+            this.attached = true;
```

Comment:
Not related to your changes, but is this state really needed? 
If you have a parent node and you are a root custom element, then you are attached.